AutoRest/Generators/Ruby/Ruby/ClientModelExtensions.cs

```diff
@@ -263,235 +265,286 @@ public static bool HasModelTypes(this ServiceClient client)
         }
 
         /// <summary>
-        /// Generates Ruby code in form of string for deserializing object of given type.
+        /// Determines whether one composite type derives directly or indirectly from another.
         /// </summary>
-        /// <param name="type">Type of object needs to be deserialized.</param>
-        /// <param name="scope">Current scope.</param>
-        /// <param name="valueReference">Reference to object which needs to be deserialized.</param>
-        /// <returns>Generated Ruby code in form of string.</returns>
-        public static string DeserializeType(
-            this IType type,
-            IScopeProvider scope,
-            string valueReference)
+        /// <param name="type">Type to test.</param>
+        /// <param name="possibleAncestorType">Type that may be an ancestor of this type.</param>
+        /// <returns>true if the type is an ancestor, false otherwise.</returns>
+        public static bool DerivesFrom(this CompositeType type, CompositeType possibleAncestorType)
         {
-            var composite = type as CompositeType;
-            var sequence = type as SequenceType;
-            var dictionary = type as DictionaryType;
-            var primary = type as PrimaryType;
-            var enumType = type as EnumType;
+            return
+                type.BaseModelType != null &&
+                (type.BaseModelType.Equals(possibleAncestorType) ||
+                 type.BaseModelType.DerivesFrom(possibleAncestorType));
+        }
 
+        public static string ConstructMapper(this IType type, string serializedName, IParameter parameter, bool isPageable, bool expandComposite)
```

Comment:
Please add comments to indicate purpose for the method and explanation of paramaters and return type.